discovery-infra/test_infra/helper_classes/hypershift.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import subprocess
+from base64 import b64decode
+
+import waiting
+from kubernetes.client import ApiClient, CoreV1Api, CustomObjectsApi, V1NodeList
+from test_infra import utils
+from test_infra.helper_classes.kube_helpers import Secret, create_kube_api_client
+
+logger = logging.getLogger(__name__)
+
+HYPERSHIFT_DIR = "capi/hypershift/"
+DEFAULT_WAIT_FOR_NODES_TIMEOUT = 5 * 60
+
+
+class HyperShift:
+    """
+    Hypershift resource that allow interaction with hypershift CLI, nodepool and hypershift cluster kubeapi-server.
+    """
+
+    def __init__(self, name: str):
+        self.name = name
+        self.kubeconfig_path = ""
+        self.hypershift_cluster_client = None
+
+    def create(self, pull_secret_file: str, ssh_key: str = ""):
+        # TODO: change this to create namespace, pull-secret, hostedCluster and nodePool
+        logger.info(f"Creating HyperShift cluster {self.name}")
+        cmd = f"./bin/hypershift create cluster agent --pull-secret {pull_secret_file} --name {self.name}"
+        if ssh_key:
+            cmd += f" --ssh-key {ssh_key}"
+        subprocess.check_call(cmd, shell=True, cwd=HYPERSHIFT_DIR)
```

Comment:
prefer if you can use "run_command" from utils

Reply:
:disappointed:

Reply:
Done